apps/ftu/js/late_customization.js

```diff
@@ -0,0 +1,657 @@
+          };
+          req.onsuccess = req.onerror = onOutcome;
+        });
+      }).catch(() => {
+          this.debug('Lost connection while attempting install of ' +
+                    appData.manifestURL);
+      });
+    },
+
+    enqueAppInstall: function(app) {
+      var installQueue = this._installQueue || (this._installQueue = []);
+      installQueue.push(app);
+      this.debug('enqueAppInstall, queue: ', app.manifestURL);
+
+      var currentInstallAttempt = this._installAttempt || Promise.resolve();
+      currentInstallAttempt.then(req => {
+        if (req && req.error) {
+          console.warn('install error:', req.error);
+        }
+        if (this._installQueue.length) {
+          this._installAttempt =
+            this.installApp(this._installQueue.shift());
+        }
+      });
+    },
+
+    debug: function() {
+      if (!this.DEBUG) {
+        return;
+      }
+      var args = Array.from(arguments);
+      args.unshift('LateCustomizationPanel');
+      console.log.apply(console, args);
+    },
+
+    getManifestUrl: function(url, infos={}) {
+      var apiURL = new URL(url);
+      var params = new URLSearchParams(apiURL.search.substr(1));
+      Object.keys(infos).forEach(key => {
+        params.set(key, infos[key]);
+      });
+      apiURL.search = params.toString();
+      return apiURL.href;
+    },
+
+    fetchManifest: function() {
+      var url = this.getManifestUrl(this._customizationManifestUrl,
+                                    this._operatorInfo);
+      if (!url) {
+        return Promise.reject(new Error('No manifest URL available'));
+      }
+
+      return new Promise((resolve, reject) => {
+        var xhr = new XMLHttpRequest({ mozSystem: true, mozAnon: true });
+        xhr.onerror = (err) => {
+          reject(new Error('Failed to fetch manifest: ' + url +
+                           ', status: ' + xhr.statusText));
+        };
+        xhr.onload = () => {
+          this.debug('onload, manifest loaded: ',
+                      xhr.response);
+          var responseData;
+          if (xhr.responseText) {
+            var ex;
+            try {
+              responseData = JSON.parse(xhr.responseText);
+            } catch (e) {
+              console.warn('exception parsing json: ', e);
+              ex = e;
+            }
+            if (ex || !responseData) {
+              console.warn('parse error: ', ex, null);
+              return reject(new Error('Invalid manifest response from ' + url));
+            }
+          }
+          var detail = responseData && responseData.detail;
+          if (!responseData ||
+              detail && detail.toLowerCase().includes('error')) {
+            reject(new Error('Empty/error response received from: ' + url));
+          } else {
+            resolve(responseData);
+          }
+        };
+        xhr.overrideMimeType('application/json');
+        this.debug('fetchManifest: ' + url);
+        xhr.open('GET', url, true);
```

Comment:
You can also use LazyLoader.getJSON to load JSON over XHR, I think that's recommended.